### ISSUE TITLE
[@types/react-redux]

### ISSUE THREAD

**Issue by karevn** [DefinitelyTyped/DefinitelyTyped#26903]
react-redux 6.0.3 version does not compile as it uses the `Dispatch` interface imported from redux. As this interface is generic, it requires generic params not provided by react-redux.

**Comment by crazyx13th** [comment]
hey there, same here I think. I got since version 6.0.0 this error: 

```
TS2322: Type '{}' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Footer> & Readonly<{ children?: ReactNode; }> & Re...'.
  Type '{}' is not assignable to type 'Readonly<Props>'.
    Property 'dispatch' is missing in type '{}'.
```

thx!

**Comment by shincysl** [comment]
sorry, my mistake. My error was Dispatch was removed. Please don't do this in a patch update..  Examples such as https://github.com/Microsoft/TypeScript-React-Starter is still using Dispatch from react-redux.

**Comment by go2ready** [comment]
Agree, the type was removed in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/26360/files#diff-ee7ace1841a23c70c0c2b663cfae1cb7L38

Which I think the contributor wasn't realized that people actually use these types if this is going to be the future, what should be the alternative for all these types that get removed?

**Comment by SpinLock79** [comment]
You have to change  import { Dispatch } from "react-redux"; to  import { Dispatch } from "redux";

**Comment by OliverJAsh** [comment]
How did these types get into a broken state? PRs shouldn't be merged if they cause the build to fail?

I tried fixing this in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/27018.

**Comment by OliverJAsh** [comment]
Ah, turns out my `types/react-redux/node_modules` was outdated.

**Comment by jeremejevs** [comment]
Sorry, refactored the types away without realizing that people could be depending on them. But yes, this is addressable by importing them directly from Redux.

The root of the Dispatch incompatibility problem, I believe, is that [there is no generic default in Redux v3](https://github.com/reduxjs/redux/blob/v3.7.2/index.d.ts#L107-L109). This didn't get caught by the tests and the usage in my own projects because it's Redux v4 everywhere. That's why @SpinLock79's suggestion doesn't work for some. You're on v3, @karevn, right?

Honestly, don't know what is the best course of action here. These typings are Redux-v4-ready as of [this PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/25109), and I think it's just luck that they were still working with v3. If we knew about this backward incompatibility, 6.0.3 would maybe be released as 7.0.0, but is there any point in an undo after a whole week?

A potential workaround for v3 users is to supply `AnyAction` as the generic parameter of `Dispatch`. If that doesn't help, then pinning to `6.0.2` should be fine too, the change in `6.0.3` is negligible, if you weren't bothered by the problem it fixes.

Input is welcome from other maintainers though, this is a somewhat tricky situation.

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
